elephant/test/test_spike_train_synchrony.py

```diff
@@ -190,9 +192,255 @@ def test_spike_contrast_with_Izhikevich_network_auto(self):
+
+        sampling_rate = 1 / pq.s
+
+        spiketrains = [neo.SpikeTrain([1, 1, 5, 10, 13, 16, 17, 19] * pq.s,
+                                      t_stop=21*pq.s),
+                       neo.SpikeTrain([1, 4, 7, 9, 12, 14, 16, 20] * pq.s,
+                                      t_stop=21*pq.s)]
+
+        correct_annotations = np.array([[3, 3, 2, 2, 3, 3, 3, 2],
+                                        [3, 2, 1, 2, 3, 3, 3, 2]])
+
+        self._test_template(spiketrains, correct_annotations, sampling_rate,
+                            spread=1, mode='delete', binary=False,
+                            in_place=True, deletion_threshold=3)
+
+    def test_extract(self):
+
+        # test synchrofact search taking into account adjacent bins
+        # this requires an additional loop with shifted binning
+
+        sampling_rate = 1 / pq.s
+
+        spiketrains = [neo.SpikeTrain([1, 5, 9, 11, 13, 20] * pq.s,
+                                      t_stop=21*pq.s),
+                       neo.SpikeTrain([1, 4, 7, 12, 16, 18] * pq.s,
+                                      t_stop=21*pq.s)]
+
+        correct_annotations = np.array([[2, 2, 1, 3, 3, 1],
+                                        [2, 2, 1, 3, 1, 1]])
+
+        self._test_template(spiketrains, correct_annotations, sampling_rate,
+                            spread=1, mode='extract', in_place=True,
+                            deletion_threshold=2)
+
+    def test_binning_for_input_with_rounding_errors(self):
+
+        # a test with inputs divided by 30000 which leads to rounding errors
+        # these errors have to be accounted for by proper binning;
+        # check if we still get the correct result
+
+        sampling_rate = 30000 / pq.s
+
+        spiketrains = [neo.SpikeTrain(np.arange(1000) * pq.s / 30000,
+                                      t_stop=.1 * pq.s),
+                       neo.SpikeTrain(np.arange(2000, step=2) * pq.s / 30000,
+                                      t_stop=.1 * pq.s)]
+
+        first_annotations = np.ones(1000)
+        first_annotations[::2] = 2
+
+        second_annotations = np.ones(1000)
+        second_annotations[:500] = 2
+
+        correct_annotations = np.array([first_annotations,
+                                        second_annotations])
+
+        self._test_template(spiketrains, correct_annotations, sampling_rate,
+                            spread=0, mode='delete', in_place=True,
+                            deletion_threshold=2)
+
+    def test_correct_transfer_of_spiketrain_attributes(self):
+
+        # for delete=True the spiketrains in the block are changed,
+        # test if their attributes remain correct
+
+        sampling_rate = 1 / pq.s
+
+        spiketrain = neo.SpikeTrain([1, 1, 5, 0] * pq.s,
+                                    t_stop=10 * pq.s)
+
+        block = neo.Block()
+
+        channel_index = neo.ChannelIndex(name='Channel 1', index=1)
+        block.channel_indexes.append(channel_index)
```

Comment:
There's also still channelIndexes here...

Reply:
right, we should get rid of old neo representation support in this PR

Reply:
Fixed.